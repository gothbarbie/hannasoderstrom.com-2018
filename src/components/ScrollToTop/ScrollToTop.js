// @flow
import { Component } from 'react'
import { withRouter } from 'react-router-dom'

type Props = {
  location: string,
  children: Object,
}

export class ScrollToTop extends Component<Props> {
  componentDidUpdate (prevProps: Props) {
    if (this.props.location !== prevProps.location) {
      window.scrollTo(0, 0)
    }
  }

  render () {
    return this.props.children
  }
}

export default withRouter(ScrollToTop)
