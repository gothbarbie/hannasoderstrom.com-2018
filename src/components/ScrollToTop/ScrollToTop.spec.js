import React from 'react'
import { Route, BrowserRouter as Router } from 'react-router-dom'
import { mount, shallow } from 'enzyme'
import ScrollToTop from './index'

describe('components/ScrollToTop', () => {
  let component

  it('renders', () => {
    component = mount(
      <Router>
        <Route>
          <ScrollToTop>
            <div>Test</div>
          </ScrollToTop>
        </Route>
      </Router>
    )
    expect(component).toMatchSnapshot()
  })
})
