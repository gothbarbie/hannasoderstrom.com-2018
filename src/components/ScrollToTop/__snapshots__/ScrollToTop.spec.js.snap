// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ScrollToTop renders 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Route>
      <withRouter(ScrollToTop)>
        <Route
          render={[Function]}
        >
          <ScrollToTop
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <div>
              Test
            </div>
          </ScrollToTop>
        </Route>
      </withRouter(ScrollToTop)>
    </Route>
  </Router>
</BrowserRouter>
`;
